pkg/zone/feature/access/group.go

```diff
@@ -0,0 +1,177 @@
+package access
+
+import (
+	"context"
+
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/smart-core-os/sc-golang/pkg/masks"
+	"github.com/vanti-dev/sc-bos/pkg/gen"
+	"github.com/vanti-dev/sc-bos/pkg/util/pull"
+	"github.com/vanti-dev/sc-bos/pkg/zone/feature/run"
+)
+
+type Group struct {
+	gen.UnimplementedAccessApiServer
+
+	client gen.AccessApiClient
+	names  []string
+
+	logger *zap.Logger
+}
+
+func (g *Group) GetLastAccessAttempt(ctx context.Context, request *gen.GetLastAccessAttemptRequest) (*gen.AccessAttempt, error) {
+	fns := make([]func() (*gen.AccessAttempt, error), len(g.names))
+	for i, name := range g.names {
+		request := proto.Clone(request).(*gen.GetLastAccessAttemptRequest)
+		request.Name = name
+		fns[i] = func() (*gen.AccessAttempt, error) {
+			return g.client.GetLastAccessAttempt(ctx, request)
+		}
+	}
+	allRes, allErrs := run.Collect(ctx, run.DefaultConcurrency, fns...)
+
+	err := multierr.Combine(allErrs...)
+	if len(multierr.Errors(err)) == len(g.names) {
+		return nil, err
+	}
+
+	if err != nil {
+		if g.logger != nil {
+			g.logger.Warn("some access implementors failed to get", zap.Errors("errors", multierr.Errors(err)))
+		}
+	}
```

Comment:
@dloveen-v Possibly, I understand that if we can't get all the devices in a zone then the last access attempt might not be technically correct as we don't have all the info. That being said, this just follows the same pattern as the other zone features, (see pkg/zone/feature/airquality/group.go:48, pkg/zone/feature/electric/group.go:48 etc.) so I think we should keep it the same for all and then think about how we handle this for all the zones, initial thought:

If we are going to do something here when we can't fetch all the nodes in a zone, it might be better for the Response to contain a confidence value or something similiar? I.e. if we can get 9/10 air quality sensors we can be quite confident the average is correct, but if we only get readings from 2 we are not that confident in our calculation, what do you think?

Reply:
Good point. If at least 1 Sensor returns, it should return that as best effort.

I don't think we want to embed confidence in traits that don't support it explicitly. Unless we change zones to have a definition of confidence.